### DISCUSSION TITLE
a good workflow to build rez packages "on demand"?

### DISCUSSION THREAD

**Discussion by bryab** [AcademySoftwareFoundation/rez#1667] [Q&A]
I am wondering if anyone has suggestions on the correct workflow for my problem, which i'll try to describe as concisely as possible.

- I do not have a shared file server - or at least, not one that can store large amounts of data, due to low bandwidth.  Storing a huge repository of Rez packages is not realistically possible.
- Most of the Rez packages can be generated via a build script, using sources accessible on the internet (e.g. pypi).

I'd like a workflow like this:  

At the time when the user attempts to create a rez context, any dependencies required for the context are automatically built and stored in the user's local packages folder.  The `rez env` command hangs until this is done, and the context is created after all the builds have finished.

Thank you for any suggestions!

**Comment by maxnbk** [comment]
This is basically (or close-to) the goal of https://github.com/AcademySoftwareFoundation/rez/issues/673 .
We're up to REP-002.007 , and some interested parties have considered tackling REP-002.008 which is a pretty core feature all on its own, in addition to empowering this sort of workflow.
It also shares a lot of commonalities with @anderslanglands [rez-cook](https://github.com/anderslanglands/rez-cook) project.
